pkg/combination/combination.go

```diff
@@ -86,6 +86,17 @@ func (cs *stream) nextIterativeCombination() (map[string]string, error) {
 	// comboList is a variable that holds a list of combinations to be returned.
 	comboList := map[string]string{}
 
+	// Edge case: 1 parameter
```

Comment:
This will work but from a readability stand-point this function is getting a bit out of hand. I'd like to see some comments and have the logic broken up a bit where makes sense. For example, a couple of these for loops could potentially go into other functions or refactored.

Reply:
I think that's a great idea. Would we want the readability improved in this PR or in a separate issue?

Reply:
Let's make it a follow-up.